peerconnection.go

```diff
@@ -869,7 +869,14 @@ func (pc *PeerConnection) CreateAnswer(options *AnswerOptions) (SessionDescripti
 
 	connectionRole := connectionRoleFromDtlsRole(pc.api.settingEngine.answeringDTLSRole)
 	if connectionRole == sdp.ConnectionRole(0) {
-		connectionRole = connectionRoleFromDtlsRole(defaultDtlsRoleAnswer)
+        dtlsRole := dtlsRoleFromRemoteSDP(remoteDesc.parsed) 
+        if dtlsRole == DTLSRoleAuto {
+            connectionRole = connectionRoleFromDtlsRole(defaultDtlsRoleAnswer)
+        } else if dtlsRole == DTLSRoleClient {
+            connectionRole = connectionRoleFromDtlsRole(DTLSRoleServer)
+        } else if dtlsRole == DTLSRoleServer {
+            connectionRole = connectionRoleFromDtlsRole(DTLSRoleClient)
+        }
```

Comment:
nit: we currently "could use tagged switch on enum"  QF1003  lint because it cases conflicts with other linters, but I think it's cleaner to use a switch here
```suggestion
		switch dtlsRole {
		case DTLSRoleAuto:
			connectionRole = connectionRoleFromDtlsRole(defaultDtlsRoleAnswer)
		case DTLSRoleClient:
			connectionRole = connectionRoleFromDtlsRole(DTLSRoleClient)
		case DTLSRoleServer:
			connectionRole = connectionRoleFromDtlsRole(DTLSRoleServer)
		default:
		}
```